src/app/content/practiceQuestions/components/Question.tsx

```diff
@@ -64,9 +66,12 @@ const Question = () => {
   const showCorrect = showCorrectState === question;
 
   const onSubmit = (e: React.FormEvent) => {
-    // TODO: Add support for handling Finish button
     e.preventDefault();
     dispatch(setAnswer({ answer: selectedAnswer, questionId: question.uid }));
+    // handling Finish button
+    if (isFinalQuestion && isSubmitted) {
+      dispatch(finishQuestions());
+    }
```

Comment:
In this case, `setAnswer` will be called after clicking `Finish`
This statement should be probably directly after the `e.preventDefault();` and `return`

Reply:
So you mean something like this, right?
 `e.preventDefault()`
`if (...) {dispatch(finishQuestions()); return;}`
`disptach(setAnswer(...));`
or
 `e.preventDefault()`
`if (...) {dispatch(finishQuestions());}`
`else {disptach(setAnswer(...));}`

Reply:
yes, both options are okay
I like using `return` for this type of situations. I'm not sure if this is in this project but we have somewhere a lint rule saying no to use `else` if not necessary